### PR TITLE
Don't litter /tmp with test artifacts.

### PR DESCRIPTION
**Pull request by josepht**
* Make the download location for source/origins mockable to
  make cleaning up after tests easier.

LP:#1624099

### CODE REVIEW

File: snapcraft/tests/test_parser.py [hunk 1]
```diff
@@ -66,6 +71,15 @@ def tearDown(self):
         except FileNotFoundError:
             pass
 
+    def setUp(self):
+        super().setUp()
+        tempdir = tempfile.mkdtemp()
```
Comment (sergiusens) [inline snapcraft/tests/test_parser.py]: why not use `fixtures.TempDir` instead?

File: snapcraft/tests/test_parser.py [hunk 2]
```diff
@@ -66,6 +71,15 @@ def tearDown(self):
...
+        patcher = mock.patch('snapcraft.internal.parser._get_base_dir')
+        base_dir = patcher.start()
+        base_dir.return_value = tempdir
```
Comment (sergiusens) [inline snapcraft/tests/test_parser.py]: if using `fixtures.TempDir` then this would be `tempdir.path`

Comment (josepht) [inline snapcraft/tests/test_parser.py]: I've converted the parser tests to use fixtures.TempDir().  I'm not 100% certain I'm handling the cleanUp correctly so please review that part carefully.

File: snapcraft/tests/test_parser.py [hunk 3]
```diff
@@ -804,7 +818,8 @@ def test_missing_snapcraft_yaml_without_debug(self, mock_get):
         self.assertEqual(0, _get_part_list_count())
 
     @mock.patch('snapcraft.internal.sources.get')
-    def test_wiki_with_fake_origin_with_bad_snapcraft_yaml(self, mock_get):
+    def test_wiki_with_fake_origin_with_bad_snapcraft_yaml(
+            self, mock_get):
```
Comment (sergiusens) [inline snapcraft/tests/test_parser.py]: why add a line break here?

File: snapcraft/tests/test_parser.py [hunk 4]
```diff
@@ -821,8 +836,10 @@ def test_wiki_with_fake_origin_with_bad_snapcraft_yaml(self, mock_get):
 parts: [somepart]
 """.format(origin_url=origin_url))
 
-        origin_dir = os.path.join(BASE_DIR, _encode_origin(origin_url))
+        origin_dir = os.path.join(parser._get_base_dir(),
+                                  _encode_origin(origin_url))
         os.makedirs(origin_dir, exist_ok=True)
+        self.addCleanup(shutil.rmtree, origin_dir)
```
Comment (sergiusens) [inline snapcraft/tests/test_parser.py]: is this needed?

File: snapcraft/tests/test_parser.py [hunk 5]
```diff
@@ -851,8 +868,10 @@ def test_wiki_with_fake_origin(self, mock_get):
...
```
Comment (sergiusens) [inline snapcraft/tests/test_parser.py]: ditto

File: snapcraft/tests/test_parser.py (hunk 3 above)
Comment (josepht) [inline snapcraft/tests/test_parser.py]: Oops, I was doing some extra mocking in there during development.  Fixed.

File: snapcraft/tests/test_parser.py (hunk 4 above)
Comment (josepht) [inline snapcraft/tests/test_parser.py]: Nope, removed.

File: snapcraft/tests/test_parser.py (hunk 5 above)
Comment (josepht) [inline snapcraft/tests/test_parser.py]: ditto ;)

**Comment by josepht** [comment]
@sergiusens I've made the changes you've requested.

**Comment by elopio** [comment]
:+1:

**Pull request merged**